app/jobs/search_keyword_job.rb

```diff
@@ -5,22 +5,36 @@ class SearchKeywordJob < ApplicationJob
 
   def perform(search_result_id)
     search_result = SearchResult.find(search_result_id)
-    html_code = Search::SearchService.new(
-      keyword: search_result.keyword,
-      search_engine: search_result.search_engine
-    ).search
+    html_code = search(search_result.keyword, search_result.search_engine)
 
-    return mark_as_failed(search_result) unless html_code
+    result = parse(search_result.search_engine, html_code)
 
-    mark_as_completed(search_result, html_code)
-  rescue IcRailsHueyToby::Errors::SearchServiceError
+    mark_as_completed(search_result, html_code, result)
+  rescue IcRailsHueyToby::Errors::SearchServiceError, IcRailsHueyToby::Errors::ParseServiceError
     mark_as_failed(search_result)
   end
 
   private
 
-  def mark_as_completed(search_result, html_code)
-    search_result.update html_code: html_code, status: :completed
+  def search(keyword, search_engine)
+    Search::SearchService.new(
+      keyword: keyword,
+      search_engine: search_engine
+    ).search
+  end
+
+  def parse(search_engine, html_code)
+    Parse::ParseService.new(
+      search_engine: search_engine,
+      html_code: html_code
+    ).parse
+  end
+
+  def mark_as_completed(search_result, html_code, result)
```

Comment:
This method has an envy issue and does not do what its name says. It is best to leave as it was before (so just updating the stattus) but have another method to add the search result to the `search_result` record, e.g., `set_parsing_result(search_result, html_code, result)`

Reply:
Fixed [7c89c40](https://github.com/nimblehq/ic-rails-huey-toby/pull/37/commits/7c89c401ec959f8b86dbe55a5dbda7e802f15af0)